src/main/java/com/sofia/uni/fmi/travelly/controller/TripController.java

```diff
@@ -18,17 +17,21 @@
 @RequestMapping("trips")
 public class TripController {
 
-    private TripService tripService;
-    private TripMapper tripMapper;
-    private ItemMapper itemMapper;
+    private final TripService tripService;
+    private final TripMapper tripMapper;
+    private final ItemService itemService;
 
-    public TripController(final TripService tripService, final TripMapper tripMapper) {
+    private final ItemMapper itemMapper;
+
+    public TripController(TripService tripService,TripMapper tripMapper, ItemService itemService, ItemMapper itemMapper) {
         this.tripService = tripService;
         this.tripMapper = tripMapper;
+        this.itemService = itemService;
+        this.itemMapper = itemMapper;
     }
 
     @GetMapping("{tripId}")
-    public TripDto getTripById(@RequestParam Long tripId) {
+    public TripDto getTripById(@PathVariable Long tripId) {
```

Comment:
@RequestParam is deprecated in our project so @PathVariable notation is the right one